physics/gjk-raycast.lisp

```diff
@@ -424,16 +428,27 @@
                 (when (/= i last-updated)
                   (incf stuck))))
              ;; check if we are done
-          while (and (<= (* epsilon maxdist) (vsqrlength v))
+          while (and (<= epsilon*maxdist (vsqrlength v))
                      (<= stuck 3))
           do ;; update the simplex and find new direction
-             (setf dim (signed-volumes dim s0 s1 s2 s3 v))
+             (setf dim (signed-volumes dim s0 s1 s2 s3 v epsilon*maxdist))
           while (< dim 4) ;; current point is inside simplex (or on
                           ;; wrong side of simplex), so can't make any
                           ;; more progress.
                           ;; TODO: make sure this returns useful
                           ;; result, and/or see if it can be improved?
-
+          do (setf epsilon*maxdist
+                   ;; we test v after updating tt since a final
+                   ;; adjustment to that improves results
+                   ;; slightly. but we need to use the values
+                   ;; corresponding to v to determine cutoff
+                   ;; tolerance, so calculate that here
+                   (* epsilon
+                      (case dim
+                        (0 0)
+                        (1 (plength s0))
+                        (2 (max (plength s0) (plength s1)))
+                        (3 (max (plength s0) (plength s1) (plength s2))))))
```

Comment:
might want this to be an `ecase`?

Reply:
it is declared UB8, and tested < 4 at that point, so shouldn't matter